python/ray/serve/_private/deployment_state.py

```diff
@@ -2032,14 +2098,25 @@ def scale_deployment_replicas(
                 logger.info(f"Adding {to_add} replica{'s' * (to_add>1)} to {self._id}.")
                 for _ in range(to_add):
                     replica_id = ReplicaID(get_random_string(), deployment_id=self._id)
+
+                    # Assign rank during replica creation (startup process)
+                    assigned_rank = self._get_next_available_rank()
+                    self._replica_ranks[replica_id.unique_id] = assigned_rank
+
+                    logger.info(
```

Comment:
nit: move to `DeploymentReplica.start`?

Reply:
maybe this comment is not relevant after moving to rank manager?